homeassistant/components/voip/assist_satellite.py

```diff
@@ -149,26 +173,110 @@ async def async_set_configuration(
 
+    # -------------------------------------------------------------------------
+    # RTP/RTCP port selection
+    # -------------------------------------------------------------------------
+    def _choose_rtp_ports(self) -> tuple[int, int]:
+        rtp_port = RTP_PORT_RANGE_START
+        while True:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            sock.setblocking(False)
+
+            # Bind to a random UDP port
+            sock.bind(("", 0))
+            _, rtp_port = sock.getsockname()
+
+            # Close socket to free port for reuse
+            sock.close()
+
+            # Check that the next port up is available for RTCP
+            rtcp_port = rtp_port + 1
+            sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+            try:
+                sock.bind(("", rtcp_port))
+
+                # Will be opened again below
+                sock.close()
+
+                # Found our ports
+                break
+            except OSError:
+                # RTCP port is taken
+                pass
+
+        return rtp_port, rtcp_port
+
+    async def async_announce(
+        self, announcement: assist_satellite.AssistSatelliteAnnouncement
+    ) -> None:
+        """Announce media on the phone. First call and wait for phone.
+
+        Before calling the phone, we should check to make sure there is not
+        already an ongoing call. After calling, should block until the
+        announcement is done playing.
+        """
+        _LOGGER.debug(
+            "Calling phone for announcement (message=%s, media_id=%s, media_id_source=%s)",
+            announcement.message,
+            announcement.media_id,
+            announcement.media_id_source,
+        )
+        rtp_port, rtcp_port = self._choose_rtp_ports()
+        source = self.hass.data[DOMAIN].protocol.local_endpoint
+        destination = SipEndpoint(self.voip_device.voip_id)
+
+        await self._pipeline_task_queue.put(
+            self._run_announce_pipeline(announcement.media_id)
+        )
+
+        # Check to see if there is an existing call, if not call out to the phone
+        if self.transport is None:
+            _LOGGER.debug("No existing call, making outgoing call to %s", destination)
+            self.hass.data[DOMAIN].protocol.outgoing_call(source, destination, rtp_port)
+
     # -------------------------------------------------------------------------
     # VoIP
     # -------------------------------------------------------------------------
 
+    def connection_made(self, transport):
+        """Server is ready."""
+        # If a connection is made with nothing in the queue, we must have
+        # received a call and we start with the regular pipeline
+        _LOGGER.debug("Assist satellite connection made")
+        super().connection_made(transport)
+        if self._pipeline_task_queue.empty():
+            self._default_pipeline_task = asyncio.create_task(
```

Comment:
Seems like there should be a better way to handle this but it seems to be working.

Reply:
I think I solved this with put_nowait.